stock_move_line_auto_fill/models/stock_move.py

```diff
@@ -33,22 +42,16 @@ def _action_assign(self, force_qty=False):
         is called for a new lines.
         """
         res = super()._action_assign(force_qty=force_qty)
-        for line in self.filtered(
+        for move in self.filtered(
             lambda m: m.state
             in ["confirmed", "assigned", "waiting", "partially_available"]
         ):
-            if (
-                line._should_bypass_reservation()
-                or not line.picking_id.auto_fill_operation
-            ):
-                return res
-            lines_to_update = line.move_line_ids.filtered(
+            if move._should_bypass_reservation():
+                continue
+            for move_line in move.move_line_ids.filtered(
                 lambda l: l.qty_done != l.reserved_uom_qty
-            )
-            for move_line in lines_to_update:
-                if (
-                    not line.picking_id.picking_type_id.avoid_lot_assignment
-                    or not move_line.lot_id
-                ):
-                    move_line.qty_done = move_line.reserved_uom_qty
+            ):
+                move_line_vals = {"lot_id": move_line.lot_id}
+                if move._check_auto_fill_conditions(move_line_vals):
```

Comment:
This would cause a performance regression. Keep the original structure and just call `_get_auto_fill_flag()` and `_get_avoid_lot_assignment_flag()` where applicable.